hazelcast/src/main/java/com/hazelcast/wan/ReplicationEventObject.java

```diff
@@ -16,8 +16,28 @@
 
 package com.hazelcast.wan;
 
+import com.hazelcast.nio.serialization.Data;
+
 /**
  * Marker interface for WAN replication messages
```

Comment:
Seems this is not a marker any more.

And also cant we provide this event counting functionality out-of-the-box(assuming this interface is implemented by the end user since it is public). Is this interface really required?

Reply:
You're right, I should move it out of the public interface. 
The new methods were in the interface to avoid casting to a specific class when incrementing stats. I'll try to find a different way.

Reply:
I thought about it and we can freely add methods to this interface since it is not implemented by the user. I changed the javadoc not to say that it's a marker interface. I've kept the new methods which allow me to use polymorhism and have each event (and future events) increment its own count, instead of having a big `if-else` chain with `instanceof`.

OTOH when adding the counter removal on map/cache destroy, I found the current design unsatisfactory as the counters were located in classes which were supposed to be implemented by the user (e.g. implementations of `WanReplicationPublisher`). This is why I've reorganised it a bit and now all counters are located in the wan service.

Please take a look.